src/prommis/nanofiltration/tests/test_diafiltration.py

```diff
@@ -0,0 +1,119 @@
+#####################################################################################################
+# “PrOMMiS” was produced under the DOE Process Optimization and Modeling for Minerals Sustainability
+# (“PrOMMiS”) initiative, and is copyright (c) 2023-2024 by the software owners: The Regents of the
+# University of California, through Lawrence Berkeley National Laboratory, et al. All rights reserved.
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license information.
+#####################################################################################################
+"""
+Tests the diafiltration flowsheet
+"""
+
+from pyomo.environ import value
+
+from idaes.core.util.model_diagnostics import DiagnosticsToolbox
+from idaes.models.unit_models import MSContactor
+
+import pytest
+
+from prommis.nanofiltration.diafiltration import (
+    add_costing,
+    add_product_constraints,
+    build_model,
+    main,
+)
+
+
+@pytest.mark.component
+def test_main():
+    """
+    Tests the execution of the main function in diafiltration.py
+    """
+    m = main()
```

Comment:
Would it be reasonable to move some of the insanitation outside of your methods? Specifically so that your methods don't create/overwrite `m`, `m.fs`, and `m.fs.costing` internally. It would greatly help integrating your model into the QGESS costing if the following workflow could be run:

```
m = ConcreteModel()
m.fs = FlowsheetBlock(dynamic=False)
m.fs.costing = CouldBeAnyCostingBlock()  # for example, QGESSCosting() or DiafiltrationCosting()

# could build other blocks in the flowsheet, including other UnitModelCostingBlocks

# Then, build the DF model and costing using a single method call
from prommis.nanofiltration import main  # method to build entire model, costing, and optimization equations
main(m)  # expects m, m.fs, and m.fs.costing to already exist - also allows users to define these themselves
```

Reply:
@bpaul4 I am not quite sure what your concern is here. It looks like each test is creating a new instance of the model, so nothing should be overwritten. 

Are you suggesting the use of a `pytest.fixture` so that `main` is only run once and used for all the tests? Whilst this does reduce the amount of time spent construction models, you do need to be careful that tests do not pollute each other.

One thing I will say however - I do not think `main` should be imported from another module. If you really need this, then it probably should have a better name (such as `build_model`); however I note you already imported `build_model` which makes me wonder if `main` is really needed.

Reply:
I'm thinking more about using this model as part of a larger flowsheet where `m`, `m.fs,` and `m.fs.costing` already exist. As they are now, these methods recreate those objects.

Reply:
Also, `m.fs.costing` is defined as `DiafiltrationCosting` when `add_costing` runs, so using `QGESSCosting` means duplicating all of the costing code rather than just calling the costing method.

Reply:
For this particular file, I don't see that as a problem as this is a test file for the main code (which might be part of my confusion). It might be more useful to comment on specific parts of the actual methods on how to address any limitations that might exist.